frontend/packages/ux-editor/src/hooks/mutations/useDeleteFormContainerMutation.ts

```diff
@@ -7,33 +7,41 @@ import type { ComponentIdsChange } from 'app-shared/types/api/FormLayoutRequest'
 import { ComponentType } from 'app-shared/types/ComponentType';
 import { useUpdateBpmn } from 'app-shared/hooks/useUpdateBpmn';
 import { removeDataTypeIdsToSign } from 'app-shared/utils/bpmnUtils';
+import { getAllDescendants } from '../../utils/formLayoutUtils';
+import { useDeleteAppAttachmentMetadataMutation } from '../../hooks/mutations/useDeleteAppAttachmentMetadataMutation';
 
 export const useDeleteFormContainerMutation = (org: string, app: string, layoutSetName: string) => {
   const { layout, layoutName } = useSelectedFormLayoutWithName();
   const formLayoutsMutation = useFormLayoutMutation(org, app, layoutName, layoutSetName);
+  const deleteAppAttachmentMetadataMutation = useDeleteAppAttachmentMetadataMutation(org, app);
   const updateBpmn = useUpdateBpmn(org, app);
   return useMutation({
     mutationFn: async (id: string) => {
       const updatedLayout: IInternalLayout = ObjectUtils.deepCopy(layout);
       const componentIdsChange: ComponentIdsChange = [];
 
-      const childrenComponentIds = layout.order[id];
-      const allComponentIds = Object.keys(layout.components);
+      const childrenFormItemIds = getAllDescendants(layout, id);
+      const allFormItemIds = Object.keys(layout.components).concat(Object.keys(layout.containers));
 
-      const fileUploadComponentIds = childrenComponentIds.filter(
-        (componentId) =>
-          layout.components[componentId].type === ComponentType.FileUpload ||
-          layout.components[componentId].type === ComponentType.FileUploadWithTag,
-      );
+      const fileUploadComponentIds = childrenFormItemIds.filter((componentId) => {
+        return (
+          layout.components[componentId]?.type === ComponentType.FileUpload ||
+          layout.components[componentId]?.type === ComponentType.FileUploadWithTag
+        );
+      });
 
       if (fileUploadComponentIds.length > 0) {
+        fileUploadComponentIds.forEach(
+          async (fileUploadComponentId) =>
+            await deleteAppAttachmentMetadataMutation.mutateAsync(fileUploadComponentId),
```

Comment:
Hva skjer hvis vi bruker `mutate` i stedet for `mutateAsync` her?

Reply:
Jeg enig i at dette sannsynligvis var et unødvendig bruk av `await`/promise.
Når det er sagt, under testing av dette oppdaget jeg en mulig feil i backenden som gjør at disse ikke kan kjøres parallelt uten risiko for feil. Det ser ut til at det oppstår en race condition i backenden, noe som resulterer i tap av data.
Jeg la til en mutex lock på endepunktet med UserRequestsSynchronizationService så dette ikke skal oppstå, og endret denne koden til å bruke `mutate`.

Reply:
Bra sett!